cl/corpus_importer/management/commands/import_columbia.py

```diff
@@ -1,295 +1,480 @@
+    is_opinion_published,
+    map_opinion_types,
+    process_extracted_opinions,
+    read_xml_to_soup,
+)
+from cl.corpus_importer.utils import (
+    add_citations_to_cluster,
+    clean_body_content,
+    clean_docket_number,
+    get_court_id,
+    match_based_text,
+    update_cluster_panel,
+)
 from cl.lib.command_utils import VerboseCommand, logger
-from cl.lib.import_lib import (
-    get_courtdates,
-    get_min_dates,
-    get_min_nocite,
-    get_path_list,
+from cl.lib.crypto import sha1_of_file
+from cl.people_db.lookup_utils import (
+    extract_judge_last_name,
+    lookup_judge_by_last_name,
+)
+from cl.search.models import SOURCES, Court, Docket, Opinion, OpinionCluster
+
+logging.basicConfig(
+    filename=f"columbia_importer_{datetime.now().strftime('%Y-%m-%d_%H-%M-%S')}.txt"
 )
 
+CASE_NAME_TWEAKER = CaseNameTweaker()
 
-class Command(VerboseCommand):
-    help = (
-        "Parses the xml files in the specified directory into opinion "
-        "objects that are saved."
-    )
 
-    def add_arguments(self, parser):
-        parser.add_argument(
-            "dir",
-            nargs="+",
-            type=str,
-            help="The directory that will be recursively searched for xml "
-            "files.",
+def find_duplicates(
+    data: dict, valid_citations: list
+) -> Optional[OpinionCluster]:
+    """Check if there is a duplicate cluster
+
+    :param data: The columbia data
+    :param valid_citations: list with valid citations
+    :return: cluster match or None
+    """
+
+    docket_number = data["docket_number"] or ""
+    case_name = data["case_name"] or ""
+    case_name_short = data["case_name_short"] or ""
+    sha1 = data["xml_sha1"]
+
+    # Find a match using sha
+    opinions_by_sha = Opinion.objects.filter(sha1=sha1)
+    if opinions_by_sha.exists():
+        return opinions_by_sha.first().cluster
+
+    for citation in valid_citations:
+        xml_opinions_content = []
+        for op in data["opinions"]:
+            xml_opinions_content.append(op["opinion"])
+
+        # Replace new lines, sometimes they cause words to be combined and reduce the
+        # accuracy of the match
+        all_opinions_content = re.sub(
+            r"(\n)+", " ", " ".join(xml_opinions_content)
         )
-        parser.add_argument(
-            "--limit",
-            type=int,
-            default=None,
-            help="Limit on how many files to run through. By default will run "
-            "through all (or if `--random`, forever).",
+        all_opinions_soup = BeautifulSoup(
+            all_opinions_content, features="html.parser"
         )
-        parser.add_argument(
-            "--random",
-            action="store_true",
-            default=False,
-            help="If set, will run through the directories and files in random "
-            "order.",
+        inner_opinion_tags = all_opinions_soup.find_all()
+        if inner_opinion_tags and inner_opinion_tags[-1].name == "page_number":
```

Comment:
Some opinions have more than 1 <page_number\> tag. Why not get rid of all of those tags instead of only the last one?
For example arizona/court_opinions/documents/e574c2908e4b3f3d.xml has 3 tags